run_tests.py

```diff
@@ -53,117 +59,185 @@ def modified_environment(**kwargs):
     os.environ.update(old_env)
 
 
-def run_test(prefix, device, ioc_launcher, lewis_launcher):
+def run_test(prefix, test_module, device_launchers):
     """
-    Runs the tests for the specified IOC.
+    Runs the tests for the specified set of devices.
 
     :param prefix: the instrument prefix
-    :param device: the name of the IOC type
-    :param ioc_launcher: the ioc launcher
-    :param lewis_launcher: the lewis simulator to use; To not use use the LewisNone object
+    :param test_module: the test module
+    :param device_launchers: context manager that launches the necessary iocs and associated emulators
     """
     # Define an environment variable with the prefix in it
     # This can then be accessed elsewhere
     os.environ["testing_prefix"] = prefix
 
-    # Load the device's test module and get the class
-    m = load_module('tests.%s' % device.lower())
+    # Need to set epics address list to local broadcast otherwise channel access won't work
+    settings = {
+        'EPICS_CA_ADDR_LIST': "127.255.255.255"
+    }
```

Comment:
Why can't you use the address list that comes from config_env? I assume because you don't want the test framework looking at the gateway? If so, please add a comment to this effect.

Reply:
This isn't behaviour I've changed - I'm not quite sure why it was done in that way in the first place. I imagine you're correct about the gateway.

Github hasn't given the most helpful diff but see old version of `run_tests.py`, line 91 for where this was being done before.

Reply:
Apologies, we'll keep as is for now then.